inspirehep/modules/warnings/ext.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Extension that suppresses warnings.
+
+By default Invenio-Base enables deprecation warnings. This extension
+disables them, unless ``DEBUG_SHOW_DEPRECATION_WARNINGS`` is ``True``.
+
+This extension also disables SQLAlchemy warnings, unless the variable
+``DEBUG_SHOW_SQLALCHEMY_WARNINGS` is set to ``True``.
+
+FIXME: add an abstract way of expressing the suppression of a warning.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+import warnings
+
+from sqlalchemy.exc import SAWarning
+
+
+class INSPIREWarnings(object):
+
+    """Extension that suppresses warnings."""
+
+    def __init__(self, app=None):
+        """Initialize the extension."""
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        """Initialize the application."""
+        self.init_config(app)
+
+        if not app.config['DEBUG_SHOW_DEPRECATION_WARNINGS']:
+            warnings.filterwarnings('ignore', category=DeprecationWarning)
```

Comment:
Well, that's what this code has been doing for the past six months. They are going to break compatibility without warning anyway.